### PR TITLE
Dask gdf fixes

### PR DESCRIPTION
**Pull request by mrocklin**
This PR includes a few small fixes that were useful in getting dask-gdf to inherit nicely from dask.dataframe.  In doing so we found that the PyGDF code needed to match pandas slightly more in a few cases.  

This is a work in progress.  I would not merge this.  So far it's up for discussion only.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by mrocklin** [comment]
See also:

-  https://github.com/dask/dask/pull/4015
-  https://github.com/gpuopenanalytics/dask_gdf/pull/43

@kkraus14

**Comment by scopatz** [comment]
LGTM

File: pygdf/series.py [hunk 1]
```diff
@@ -306,6 +306,12 @@ def __mul__(self, other):
     def __rmul__(self, other):
         return self._rbinaryop(other, 'mul')
 
+    def __pow__(self, other):
+        if other == 2:
+            return self * self
+        else:
+            return NotImplemented
```
Comment (mrocklin) [inline pygdf/series.py]: This is janky.  Is there a GPU implementation somewhere that we can use?

File: pygdf/series.py [hunk 2]
```diff
@@ -688,35 +694,35 @@ def find_last_value(self, value):
...
         return self._column.mean()
 
-    def std(self, ddof=1):
+    def std(self, ddof=1, axis=None, skipna=True):
         """Compute the standard deviation of the series
         """
         return np.sqrt(self.var(ddof=ddof))
 
-    def var(self, ddof=1):
+    def var(self, ddof=1, axis=None, skipna=True):
```
Comment (mrocklin) [inline pygdf/series.py]: I probably need to add input validation to ensure that the user doesn't specify anything *except* for the defaults.   This silently produces the wrong result if they ask for something other than the defaults.

Comment (scopatz) [reply pygdf/series.py]: yes, that would be nice. Other parts of the *gdf ecosystem use simple asserts to cover the unsupported APIs. eg

```py
assert axis is None and skipna
```

File: pygdf/series.py (hunk 1 above)
Comment (kkraus14) [reply pygdf/series.py]: @mrocklin This is a WIP in https://github.com/gpuopenanalytics/libgdf/pull/94

**Comment by kkraus14** [comment]
add to whitelist

File: pygdf/series.py [hunk 3]
```diff
@@ -59,9 +60,20 @@ def from_masked_array(cls, data, mask, null_count=None):
...
     def __init__(self, data, index=None):
+        name = None
+        if isinstance(data, pd.Series):
+            from .dataframe import DataFrame
+            name = data.name
+            data = data.to_frame()
+            data.columns = ['x']
+            data = DataFrame.from_pandas(data)
+            data = data['x']
+            data.name = name
```
[the remark this answers is not in the document]
Comment (mrocklin) [reply pygdf/series.py]: Fair point.  I'll dive through and figure out how to construct a Series object manually in the morning.  Any pointers would be welcome.

[the remark this answers is not in the document]
Comment (kkraus14) [reply pygdf/series.py]: Currently a pygdf.Series can be constructed from a pandas.Series just via normal `__init__` which ends up hitting this clause: https://github.com/gpuopenanalytics/pygdf/blob/master/pygdf/columnops.py#L183

So the Pandas Series is casted to a numpy array. I think you could add a check for if it's an instance of a Pandas Series and set the name if it is and then call the `columnops.as_column` function.

**Comment by kkraus14** [comment]
Changes look good to me thus far, some of the binaryops will require a bit of refactoring once new binaryops land in libgdf, but that's to be expected.

**Comment by mrocklin** [comment]
> Changes look good to me thus far, some of the binaryops will require a bit of refactoring once new binaryops land in libgdf, but that's to be expected.

If this seems mergable to you without added burden then I would not be against merging it.  I think that most of this is defensible regardless of what happens.

**Pull request merged**